### ISSUE TITLE
Satchel / Bag Option

### ISSUE THREAD

**Issue by cseufert** [dvdoug/BoxPacker#141]
It would be great to have a way to pack into satchels as well as boxes, or really small items. Australia Post sells small, medium and large prepaid satchels.  (https://auspost.com.au/sending/satchels-and-packaging/send-in-satchel-or-envelope/flat-rate-satchels)

I can see this being fairly tricky, as the length and width get small depending on the height of the item placed inside.

Do you have any tips if I were to have a go at implementing this myself?

**Comment by dvdoug** [comment]
Yeah, to do this properly will be quite difficult - unpicking the assumptions in the code that width/height/depth are fixed would be doable, but care would need to be taken not to regress performance too much.

The main problem though would actually modelling the shape of the satchel - I'm not sure I even know what words to search for to find the appropriate formula 🤔

**Comment by spectravp** [comment]
Howdy, 

Since envelopes, satchels, etc have become common in shipping, I thought I would chime in and support the addition of this functionality. This is known as the Tea Bag problem and the math is actually pretty straightforward: https://en.wikipedia.org/wiki/Paper_bag_problem

**Comment by dvdoug** [comment]
@spectravp knowing the formal name for this problem is very helpful, thank you! Unfortunately I don't think that formula is quite sufficient for what's needed here, as a simple volume comparison isn't enough - we need to do something along the lines of calculating the (dynamic) depth of the open satchel at points along it's length so it's possible to make a decision about fit.

I see some reading ahead of me

**Comment by colinmollenhour** [comment]
As a simple workaround you could probably introduce two or more "boxes" for each envelope using predetermined dimensions based on how full it is height-wise. E.g. a 10x14 envelope might have the following rough sizes:

- 10x14x0.5
- 9x11x1.5
- 8x10x2.5
- 6x8x4.5

**Comment by noknokcody** [comment]
Any update on this? I'm looking for a way of calculating how many items fit in a satchel-like container for shipping. Perhaps interested in sponsoring the development of this feature and a couple others if possible.

**Comment by dvdoug** [comment]
Hi @noknokcody - I saw you reached out on LinkedIn, have replied to you there.

**Comment by erikkvarnbranna** [comment]
The (simple) way I solved this was. 
 - 1. Set the bag height to 0. 
 - 2. Sort the remaining items
 - 3. Add the first items height to my bag and make width and length the same amount shorter.
 - 4. Try fitting items
 - 5. If items left return to 2.

Simple and far from perfect but seems to work reasonably well.
